dotnet-desktop-guide/net/wpf/events/snippets/how-to-create-a-custom-routed-event/csharp/CodeSample.csproj

```
@@ -0,0 +1,31 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>net6.0-windows</TargetFramework>
+    <UseWPF>true</UseWPF>
+    <NeutralLanguage>en-us</NeutralLanguage>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Reference Include="WpfControlLibrary">
+      <HintPath>WpfControlLibrary.dll</HintPath>
+    </Reference>
+  </ItemGroup>
```

The proper way for this would be to reference the project directly.
```suggestion
  <ItemGroup>
    <ProjectReference Include="..\WpfControlLibraryCsharp\WpfControlLibrary.csproj" />
  </ItemGroup>
```

However, opening this project in Visual Studio when it uses a `ProjectReference` requires the solution file to have both projects listed. The `dotnet` command line doesn't care though and will still compile just fine.

This way you don't need to include the compiled `WpfControlLibrary.dll` in the folder either.

Also, if the library project isn't being used for snippets, you could just keep a single C# version of it and reference it from both VB/C# main projects.

---

In this case the library project is one of the snippets.